### PR TITLE
feat: migration from 14 to 15

### PR DESCRIPTION
**Pull request by hacdias**
My first migration ❤️ 

### Included migration tasks

- [x] https://github.com/ipfs/kubo/issues/9496
- [x] https://github.com/ipfs/kubo/issues/10005

### CODE REVIEW

File: fs-repo-14-to-15/go.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module github.com/ipfs/fs-repo-migrations/fs-repo-14-to-15
+
+go 1.18
```
Comment (Jorropo) [inline fs-repo-14-to-15/go.mod:3]:
```suggestion
go 1.20
```

File: fs-repo-14-to-15/main.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package main
+
+import (
+	mg14 "github.com/ipfs/fs-repo-migrations/fs-repo-14-to-15/migration"
+	migrate "github.com/ipfs/fs-repo-migrations/tools/go-migrate"
+)
+
+func main() {
+	m := mg14.Migration{}
```
Comment (Jorropo) [inline fs-repo-14-to-15/main.go:9]: 14 ? I don't remember if we use the previous or future version id.

Comment (hacdias) [reply fs-repo-14-to-15/main.go:9]: @Jorropo on `fs-repo-13-to-14` it says `mg13`.

File: fs-repo-14-to-15/migration/migration.go [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+					newSwarm = append(newSwarm, v)
+				}
+			}
+			addresses["Swarm"] = newSwarm
+		}
+	}
+
+	// Remove legacy Gateway.HTTPHeaders values that were hardcoded since years ago, but no longer necessary
+	// (but leave as-is if user made any changes)
+	// https://github.com/ipfs/kubo/issues/10005
```
Comment (lidel) [inline fs-repo-14-to-15/migration/migration.go:203]: :information_source: Added cleanup of `Gateway.HTTPHeaders`.  It was a long time due. 
Details and rationale in  https://github.com/ipfs/kubo/issues/10005

**Pull request merged**